src/Symfony/Component/Yaml/Parser.php

```diff
@@ -234,9 +234,11 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
                     throw new ParseException('Multiple documents are not supported.');
                 }
 
-                // 1-liner optionally followed by newline
-                $lineCount = count($this->lines);
-                if (1 === $lineCount || (2 === $lineCount && empty($this->lines[1]))) {
+                // store the last PCRE regex error code
+                $pregLastError = preg_last_error();
+
+                // 1-liner optionally followed by newline(s)
+                if (preg_match('#^.+\s*#i', $value)) {
```

Comment:
Nope I get failures:
1) Symfony\Component\Yaml\Tests\ParserTest::testEndOfTheDocumentMarker
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'foo'
+'--- %YAML:1.0
+foo
+...'

2) Symfony\Component\Yaml\Tests\ParserTest::testUnindentedCollectionException
Failed asserting that exception of type "\Symfony\Component\Yaml\Exception\ParseException" is thrown.